Tools/src/FormatCheckTool/FormatCheckTool.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Recognizers.Text;
+using Microsoft.Recognizers.Text.Choice;
+using Microsoft.Recognizers.Text.DateTime;
+using Microsoft.Recognizers.Text.Number;
+using Microsoft.Recognizers.Text.NumberWithUnit;
+using Microsoft.Recognizers.Text.Sequence;
+using Newtonsoft.Json;
+using Microsoft.Recognizers.Text.DataDrivenTests;
+
+namespace FormatCheckTool
+{
+    class FormatCheckTool
+    {
+        private static readonly String PleaseCheckStr = "Please Check it.";
+
+        static void Main(string[] args)
+        {
+            // Encoding for 'exotic' characters e.g. '€'
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+
+            ShowIntro();
+
+            while (true)
+            {
+                Console.WriteLine("Enter the specs JSON file path to be checked:");
+                var input = Console.ReadLine()?.Trim();
+                Console.WriteLine();
+
+                if (input?.ToLower(CultureInfo.InvariantCulture) == "exit")
+                {
+                    break;
+                }
+
+                // Validate input
+                if (input?.Length > 0)
+                {
+                    var pathSplit = input.Split(Path.DirectorySeparatorChar);
+                    string basicDir = pathSplit.Length > 1 ? string.Join(Path.DirectorySeparatorChar, pathSplit.Take(pathSplit.Length - 1)) : $".{Path.DirectorySeparatorChar}";
+                    string basicPath = pathSplit[pathSplit.Length - 1];
+                    if (!Directory.Exists(basicDir)) {
+                        Console.WriteLine($"File Path Error. {PleaseCheckStr}");
+                        continue;
+                    }
+
+                    string[] paths = Directory.GetFiles(basicDir, basicPath);
+                    Console.WriteLine($"Loading {paths.Length} Spec Files...");
+                    Console.WriteLine();
+
+                    foreach (string path in paths)
+                    {
+                        CheckPipeline(path);
+                    }
+
+                    Console.WriteLine();
+                }
+            }
+        }
+
+        private static void CheckPipeline(string filePath)
+        {
+            Console.WriteLine("-------------------------");
+            Console.WriteLine($"Begin to check file {filePath}.");
+            Tuple<bool, List<TestModel>> data = LoadJsonFile(filePath);
+            if (!data.Item1)
+            {
+                return;
+            }
+
+            bool stage2 = CheckBasicContent(data.Item2);
+            if (!stage2)
+            {
+                return;
+            }
+
+            bool stage3 = CheckModelResult(data.Item2);
+        }
+
+        private static Tuple<bool, List<TestModel>> LoadJsonFile(string filePath)
```

Comment:
Wouldn't it be better to define an internal class to represent this return object?